commandline.c

```diff
@@ -33,15 +34,41 @@ static void usage(const char *prgname)
 
 int main(int argc, char* argv[])
 {
+	bool print_help = false;
+	bool print_version = false;
+	int c;
 	char *iface = NULL;
 	int ret = -1;
 	bool load_embedded = true;
 	const char *progname = "ap51-flash";
-
-	if (argc >= 1)
-		progname = argv[0];
-
-	if ((argc == 2) && (strcmp("-v", argv[1]) == 0)) {
+        static struct option long_options[] = {
+                {"help",        no_argument,            0,      'h'},
+                {"version",     no_argument,            0,      'v'}
```

Comment:
You are missing the terminating entry here. Just add a line `{}`:

            static const struct option long_options[] = {
                    {"help",        no_argument,            0,      'h'},
                    {"version",     no_argument,            0,      'v'},
    		{}
            };

Reply:
No, you've added a lot of spaces in the first commit before the options. basically, all these things are incorrectly indented:

    0001-ap51-flash-Use-getopt_long-to-parse-command-line-opt.patch:+        static struct option long_options[] = {
    0001-ap51-flash-Use-getopt_long-to-parse-command-line-opt.patch:+                {"help",        no_argument,            0,      'h'},
    0001-ap51-flash-Use-getopt_long-to-parse-command-line-opt.patch:+                {"version",     no_argument,            0,      'v'},
    0001-ap51-flash-Use-getopt_long-to-parse-command-line-opt.patch:+        };
    0001-ap51-flash-Use-getopt_long-to-parse-command-line-opt.patch:+        while ((c = getopt_long(argc, argv, "hv", long_options, NULL)) != -1) {
    0001-ap51-flash-Use-getopt_long-to-parse-command-line-opt.patch:+                switch (c) {
    0001-ap51-flash-Use-getopt_long-to-parse-command-line-opt.patch:+                case 'h':
    0001-ap51-flash-Use-getopt_long-to-parse-command-line-opt.patch:+                        print_help = true;
    0001-ap51-flash-Use-getopt_long-to-parse-command-line-opt.patch:+                        ret = 0;
    0001-ap51-flash-Use-getopt_long-to-parse-command-line-opt.patch:+                        break;
    0001-ap51-flash-Use-getopt_long-to-parse-command-line-opt.patch:+                case 'v':
    0001-ap51-flash-Use-getopt_long-to-parse-command-line-opt.patch:+                        print_version = true;
    0001-ap51-flash-Use-getopt_long-to-parse-command-line-opt.patch:+                        ret = 0;
    0001-ap51-flash-Use-getopt_long-to-parse-command-line-opt.patch:+                        break;
    0001-ap51-flash-Use-getopt_long-to-parse-command-line-opt.patch:+                case '?':
    0001-ap51-flash-Use-getopt_long-to-parse-command-line-opt.patch:+                        print_help = true;
    0001-ap51-flash-Use-getopt_long-to-parse-command-line-opt.patch:+                        break;
    0001-ap51-flash-Use-getopt_long-to-parse-command-line-opt.patch:+                }
    0001-ap51-flash-Use-getopt_long-to-parse-command-line-opt.patch:+        }
    0001-ap51-flash-Use-getopt_long-to-parse-command-line-opt.patch:+        progname = argv[0];
    0001-ap51-flash-Use-getopt_long-to-parse-command-line-opt.patch:+        if (print_help) {
    0001-ap51-flash-Use-getopt_long-to-parse-command-line-opt.patch:+                usage(progname);
    0001-ap51-flash-Use-getopt_long-to-parse-command-line-opt.patch:+                goto out;
    0001-ap51-flash-Use-getopt_long-to-parse-command-line-opt.patch:+        }
    0001-ap51-flash-Use-getopt_long-to-parse-command-line-opt.patch:+        if (print_version) {


I am expecting tabs instead of 8 spaces.